src/fold/mod.rs

```diff
@@ -423,9 +423,9 @@ macro_rules! enum_fold {
 
 enum_fold!(PolarizedTraitRef[] { Positive(a), Negative(a) });
 enum_fold!(ParameterKind[T,L] { Ty(a), Lifetime(a) } where T: Fold, L: Fold);
-enum_fold!(DomainGoal[] { Implemented(a), ProjectionEq(a), Normalize(a), UnselectedNormalize(a), WellFormed(a), FromEnv(a), InScope(a) });
-enum_fold!(WellFormed[] { Ty(a), TraitRef(a), ProjectionEq(a) });
-enum_fold!(FromEnv[] { Ty(a), TraitRef(a), ProjectionEq(a) });
+enum_fold!(WhereClauseAtom[] { Implemented(a), ProjectionEq(a) });
+enum_fold!(DomainGoal[] { Holds(a), WellFormed(a), FromEnv(a), Normalize(a), UnselectedNormalize(a),
```

Comment:
as an aside, I think it was @varkor who mentioned that we should make the rustc-guide match up with this setup more closely